backend/backend/MemoryGameWithNumbers/Controllers/MemoryGameWithNumbersController.cs

```diff
@@ -1,31 +1,65 @@
 using Microsoft.AspNetCore.Mvc;
 
-namespace backend.MemoryGameWithNumbers.Controllers
+namespace backend.MemoryGameWithNumbers.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class MemoryGameWithNumbersController : ControllerBase
 {
-    [ApiController]
-    [Route("[controller]")]
-    public class MemoryGameWithNumbersController : ControllerBase
+    private static List<int?>? _correctSequence;
+
+    [HttpGet("start")]
+    public ActionResult<List<int?>> StartGame([FromQuery] int maxNumber)
     {
-        [HttpGet("start")]
-        public ActionResult<List<int?>> StartGame([FromQuery] int maxNumber)
+        Random rand = new Random();
+
+        List<int?> correctSequence = Enumerable.Range(1, maxNumber)
+                                               .Select(n => (int?)n) // Boxing happens here when converting int to int?
+                                               .ToList();
+
+        _correctSequence = new List<int?>(correctSequence);
+
+        List<int?> gridNumbers = new List<int?>(correctSequence.OrderBy(x => rand.Next()));
+
+        while (gridNumbers.Count < 16)
         {
-            List<int> generatedNumbers = new List<int>();
+            gridNumbers.Add(null);
+        }
 
-            Random rand = new Random();
+        gridNumbers = gridNumbers.OrderBy(x => rand.Next()).ToList();
 
-            generatedNumbers = Enumerable.Range(1, maxNumber)
-                                         .OrderBy(x => rand.Next())
-                                         .ToList();
+        return Ok(gridNumbers);
+    }
 
-            List<int?> gridNumbers = new List<int?>(generatedNumbers.Select(n => (int?)n));
+    [HttpPost("attempt")]
+    public ActionResult<bool> CheckAttempt([FromBody] List<int?> userAttempt)
+    {
+        if (_correctSequence == null)
+        {
+            return BadRequest("Game not started.");
+        }
 
-            while (gridNumbers.Count < 16)
+        for (int i = 0; i < _correctSequence.Count; i++)
```

Comment:
Cia galetum naudoti foreach, not sure ar yra skirtumas tarp paprasto for ir foreach

Reply:
geriau foreach. both for readability and performance